src/Microsoft.OData.Edm/Schema/CoreModel/EdmCoreModelEntityType.cs

```diff
@@ -0,0 +1,92 @@
+//---------------------------------------------------------------------
+// <copyright file="EdmCoreModelEntityType.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.OData.Edm
+{
+    /// <summary>
+    /// The built-in Edm.EntityType abstrat type in the core model.
+    /// </summary>
+    internal sealed class EdmCoreModelEntityType : EdmType, IEdmEntityType, IEdmCoreModelElement
+    {
+        /// <summary>
+        /// The core Edm.EntityType singleton.
+        /// </summary>
+        public static readonly EdmCoreModelEntityType Instance = new EdmCoreModelEntityType();
+
+        /// <summary>
+        /// Gets the kind of this type.
+        /// </summary>
+        public override EdmTypeKind TypeKind => EdmTypeKind.Entity;
+
+        /// <summary>
+        /// Gets the schema element kind of this type.
+        /// </summary>
+        public EdmSchemaElementKind SchemaElementKind => EdmSchemaElementKind.TypeDefinition;
+
+        /// <summary>
+        /// Gets the name of this type.
+        /// </summary>
+        public string Name => EdmCoreModelConstants.EntityType;
+
+        /// <summary>
+        /// Gets the namespace of this type.
+        /// </summary>
+        public string Namespace => EdmCoreModelConstants.EdmNamespace;
+
+        /// <summary>
+        /// Gets the value indicating whether or not this type is a media entity.
+        /// The Edm.EntityType has not stream.
+        /// </summary>
+        public bool HasStream => false;
+
+        /// <summary>
+        /// Gets a value indicating whether this type is abstract.
+        /// The Edm.EntityType is always an abstract type.
+        /// </summary>
+        public bool IsAbstract => true;
+
+        /// <summary>
+        /// Gets a value indicating whether this type is open.
+        /// The Edm.EntityType is always non-open type.
+        /// </summary>
+        public bool IsOpen => false;

```

Comment:
Like with Edm.ComplexType; not sure if we can assume IsOpen is either true or false.

Reply:
Synced at Standup. It's fine to make it as false for abstract type.

---
In reply to: [164583647](https://github.com/OData/odata.net/pull/1051#discussion_r164583647) [](ancestors = 164583647)